packages/core-database/src/repositories/utils/filter-rows.ts

```diff
@@ -40,13 +42,22 @@ export = <T>(rows: T[], params, filters) =>
                 if (params[elem].hasOwnProperty("from") || params[elem].hasOwnProperty("to")) {
                     let isMoreThan = true;
                     let isLessThan = true;
+                    let value = item[elem];
+
+                    if (elem === "approval") {
+                        value = delegateCalculator.calculateApproval(item);
+                    } else if (elem === "productivity") {
+                        value = delegateCalculator.calculateProductivity(item);
+                    } else if (elem === "forgedTotal") {
+                        value = delegateCalculator.calculateForgedTotal(item);
+                    }
```

Comment:
I don't like this very much in here. Another option would be to manipulate the delegates array beforehand by adding the respective properties (approval / productivity / forgedTotal). What do you say @faustbrian?

Reply:
Would need to be benchmarked what has a bigger performance hit as the manipulation would always happen while the call to this API endpoint and access to the data happens occasionally.

Reply:
It could be added into [`getLocalDelegates()`](https://github.com/ArkEcosystem/core/blob/e04596c2d8b40a8c1385c35e1ebfd21d9d5c1f74/packages/core-database/src/repositories/delegates-business-repository.ts#L18) by passing the requested params and manipulating on demand only.

Reply:
Will test how much it impacts performance if you add it there.